In [41]:
import os
import pandas as pd

os.chdir('/mnt/BioAdHoc/Groups/vd-ay/jreyna/projects/GEO-Resources/')

## Loading the query data

This data comes from a query containing data from October 16 2022 to October YY 2023

In [51]:
fn = 'results/sra/combined.meta.major_columns.renamed.tsv'
data = pd.read_table(fn, header=None)
data.columns = ['gse_id', 'gsm_id', 'srr_id', 'sample_info1', 'sample_info2', 'sample_info3', 'organism', 'num_reads']

In [52]:
data

gse_id      gsm_id       srr_id  \
0     GSE205780  GSM6226299  SRR19601532   
1     GSE196922  GSM5904681  SRR18054275   
2     GSE196922  GSM5904682  SRR18054274   
3     GSE196922  GSM5904683  SRR18054273   
4     GSE196922  GSM5904684  SRR18054272   
...         ...         ...          ...   
4317  GSE227245  GSM7093768  SRR23851551   
4318  GSE227245  GSM7093769  SRR23853107   
4319  GSE227245  GSM7093770  SRR23853108   
4320  GSE227245  GSM7093771  SRR23853109   
4321  GSE227245  GSM7093772  SRR23853110   

                                           sample_info1  \
0                H3K27ac HiChIP of wt stembryos at 120h   
1                              HCT116 Control ATAC rep1   
2                              HCT116 Control ATAC rep2   
3                                HCT116 FOXD2 ATAC rep1   
4                                HCT116 FOXD2 ATAC rep2   
...                                                 ...   
4317                                       T47D E2 Rep2   
4318  Ishikawa scRNAseq, 0,2,4 and 8 hours of e2 tre...   
4319  Ishikawa scRNAseq, 0,2,4 and 8 hours of e2 tre...   
4320  T-47D scRNAseq, 0,2,4 and 8 hours of e2 treatm...   
4321  T-47D scRNAseq, 0,2,4 and 8 hours of e2 treatm...   

                sample_info2 sample_info3      organism   num_reads  
0                   stembryo          NaN  Mus musculus   278343742  
1     Large intestine; Colon  description  Homo sapiens    36290912  
2     Large intestine; Colon  description  Homo sapiens    36156478  
3     Large intestine; Colon  description  Homo sapiens    37230632  
4     Large intestine; Colon  description  Homo sapiens    37350234  
...                      ...          ...           ...         ...  
4317                   T-47D          NaN  Homo sapiens    73525840  
4318                Ishikawa          NaN  Homo sapiens  1745416626  
4319                Ishikawa          NaN  Homo sapiens   177558440  
4320                    T47D          NaN  Homo sapiens  1814539868  
4321                    T47D          NaN  Homo sapiens   156687150  

[4322 rows x 8 columns]

In [53]:
data.shape

(4322, 8)

In total there are 4322 entries we have to sift through. I also want to know how many mention hichip at all. 

## Querying for hichip

In [54]:
# querying the metadata for mention of hichip (case-insenstive)
ss1 = data['sample_info1'].str.contains('hichip', case=False)
ss2 = data['sample_info2'].str.contains('hichip', case=False)
ss3 = data['sample_info3'].str.contains('hichip', case=False)
ss4 = data['sample_info1'].str.contains('hi-chip', case=False)
ss5 = data['sample_info2'].str.contains('hi-chip', case=False)
ss6 = data['sample_info3'].str.contains('hi-chip', case=False)
high_confidence_hichip = data.loc[ss1 | ss2 | ss3 | ss4 | ss5 | ss6]

In [55]:
high_confidence_hichip.shape

(706, 8)

Out of the 4322, 706 mention hichip. We still have to go through the remaining 3616 but at least we have a rough estimate for the number of samples that we have to process. 

In [56]:
high_confidence_hichip.gsm_id.nunique()

382